### ISSUE TITLE
Liquidity Fees are not paid to LP party at providersFeeCalculationTimeStep, instead they go to insurance as a penalty

### ISSUE THREAD

**Issue by vega-paul** [vegaprotocol/vega#9451]
See test test_liquidity_distribution
branch sla_rework_existing_1b

Test proposes and enacts a market with  sla_commitment_min_time_fraction=0.1, sla_price_range=20

Epoch is 20s, market.liquidity.providersFeeCalculationTimeStep is 19s

LP party creates two orders to reach SLA
```
"id":"7414-14","block":"AF4E74F1A6ECA80CCCF2E8465071C9FE3A5CFE7040FA72B42E75DB92A145391A","type":"BUS_EVENT_TYPE_ORDER","order":{"id":"e561d280afe982dd69d39c94e2baf561aa7e6da4c5378fa474820738ffe16116","marketId":"5cb1f66c2c363734fdd594cdb2eacf935402b0869d282db861010a7cfb0ce420","partyId":"3e70d8686e4c1d9554ac52909530a9f2c1dd3a6546c2f8121080be724d784b01","side":"SIDE_BUY","price":"100000","size":"39050000000","remaining":"39050000000","timeInForce":"TIME_IN_FORCE_GTC","type":"TYPE_LIMIT","createdAt":"1695038774855709000","status":"STATUS_ACTIVE","reference":"test_liquidity_distribution-f2370d6b","version":"1"},"version":1,"chainId":"testnet-001","txHash":"4A7528C3034F4D474A9A28D359B01A4712559D5FE4C2F1C7A469B40EF48A920C"}
{"id":"7414-17","block":"AF4E74F1A6ECA80CCCF2E8465071C9FE3A5CFE7040FA72B42E75DB92A145391A","type":"BUS_EVENT_TYPE_ORDER","order":{"id":"7a38747a384b5b3abc5b03ecfeb03f219d7bcb704baad1705b0f99d9b9461f06","marketId":"5cb1f66c2c363734fdd594cdb2eacf935402b0869d282db861010a7cfb0ce420","partyId":"3e70d8686e4c1d9554ac52909530a9f2c1dd3a6546c2f8121080be724d784b01","side":"SIDE_SELL","price":"1000000000","size":"1","remaining":"1","timeInForce":"TIME_IN_FORCE_GTC","type":"TYPE_LIMIT","createdAt":"1695038774855709000","status":"STATUS_ACTIVE","reference":"test_liquidity_distribution-e39622d3","version":"1"},"version":1,"chainId":"testnet-001","txHash":"76867FBF05B9F45BB473204B86CC7BB2EE157F91A0467CD8E1BA61A3B351046A"}
```

A trade occurs on the market to generate LP fees

```
{"id":"7468-26","block":"D977D8F37588D59DB87C1E0661F2944AC77201CE820276EF3C506046D03ECD17","type":"BUS_EVENT_TYPE_TRADE","trade":{"id":"984a9a85e72aaec4d075614e8dc757a7ef838369ae70c08b6164eba7911e604c","marketId":"5cb1f66c2c363734fdd594cdb2eacf935402b0869d282db861010a7cfb0ce420","price":"119900000","size":"100000","buyer":"085aadf0b9640d77561f1ebb058d4d46e54c7e96197fdc693e48585faf7510e3","seller":"6ad90eba20f07b1772b703f50fd1f4708d360d9cd9e4d65c2180b7adb3864dd7","aggressor":"SIDE_BUY","buyOrder":"7037e9ab944e8a279a12e53a3bcd80075cfdcc20339db3a9534844ac7fc16656","sellOrder":"39c539f62c5a4e441ac29c080b835a584a8bfe4343865754668408c34456c392","timestamp":"1695038804035182000","type":"TYPE_DEFAULT","buyerFee":{"makerFee":"4796000000000000000","infrastructureFee":"1199000000000000000","liquidityFee":"359700000000000000000","makerFeeVolumeDiscount":"0","infrastructureFeeVolumeDiscount":"0","liquidityFeeVolumeDiscount":"0","makerFeeReferrerDiscount":"0","infrastructureFeeReferrerDiscount":"0","liquidityFeeReferrerDiscount":"0"},"sellerFee":{"makerFee":"0","infrastructureFee":"0","liquidityFee":"0","makerFeeVolumeDiscount":"0","infrastructureFeeVolumeDiscount":"0","liquidityFeeVolumeDiscount":"0","makerFeeReferrerDiscount":"0","infrastructureFeeReferrerDiscount":"0","liquidityFeeReferrerDiscount":"0"}},"version":1,"chainId":"testnet-001","txHash":"B0A1CF2435E86C24B7EF3C27FCA787E2C2781892D287820211F0DB437E7E5D8A"}
```

We wait until end of epoch to check accounts, nothing added to LP general account

Two ledger events are of interest..

```
{"id":"8176-18","block":"3329CF8881290C66F6EAF46FC47ED2C398FB1868FE56C3AC7BAF2C2A32A6AFD4","type":"BUS_EVENT_TYPE_LEDGER_MOVEMENTS","ledgerMovements":{"ledgerMovements":[{"entries":[{"fromAccount":{"assetId":"c795b738b7d6636eb9c77987029a6e3301b13552cdb85f965b927752a0f995a3","type":"ACCOUNT_TYPE_BOND","owner":"3e70d8686e4c1d9554ac52909530a9f2c1dd3a6546c2f8121080be724d784b01","marketId":"5cb1f66c2c363734fdd594cdb2eacf935402b0869d282db861010a7cfb0ce420"},"toAccount":{"assetId":"c795b738b7d6636eb9c77987029a6e3301b13552cdb85f965b927752a0f995a3","type":"ACCOUNT_TYPE_INSURANCE","marketId":"5cb1f66c2c363734fdd594cdb2eacf935402b0869d282db861010a7cfb0ce420"},"amount":"0","type":"TRANSFER_TYPE_SLA_PENALTY_BOND_APPLY","timestamp":"1695039300933961000","fromAccountBalance":"11637806892395020","toAccountBalance":"390859688362193107604980"}],"balances":[{"account":{"assetId":"c795b738b7d6636eb9c77987029a6e3301b13552cdb85f965b927752a0f995a3","type":"ACCOUNT_TYPE_INSURANCE","marketId":"5cb1f66c2c363734fdd594cdb2eacf935402b0869d282db861010a7cfb0ce420"},"balance":"0"}]}]},"version":1,"chainId":"testnet-001","txHash":"3329CF8881290C66F6EAF46FC47ED2C398FB1868FE56C3AC7BAF2C2A32A6AFD4"}
{"id":"8176-21","block":"3329CF8881290C66F6EAF46FC47ED2C398FB1868FE56C3AC7BAF2C2A32A6AFD4","type":"BUS_EVENT_TYPE_LEDGER_MOVEMENTS","ledgerMovements":{"ledgerMovements":[{"entries":[{"fromAccount":{"assetId":"c795b738b7d6636eb9c77987029a6e3301b13552cdb85f965b927752a0f995a3","type":"ACCOUNT_TYPE_LP_LIQUIDITY_FEES","owner":"3e70d8686e4c1d9554ac52909530a9f2c1dd3a6546c2f8121080be724d784b01","marketId":"5cb1f66c2c363734fdd594cdb2eacf935402b0869d282db861010a7cfb0ce420"},"toAccount":{"assetId":"c795b738b7d6636eb9c77987029a6e3301b13552cdb85f965b927752a0f995a3","type":"ACCOUNT_TYPE_INSURANCE","marketId":"5cb1f66c2c363734fdd594cdb2eacf935402b0869d282db861010a7cfb0ce420"},"amount":"0","type":"TRANSFER_TYPE_SLA_PENALTY_LP_FEE_APPLY","timestamp":"1695039300933961000","fromAccountBalance":"0","toAccountBalance":"390859688362193107604980"}],"balances":[{"account":{"assetId":"c795b738b7d6636eb9c77987029a6e3301b13552cdb85f965b927752a0f995a3","type":"ACCOUNT_TYPE_INSURANCE","marketId":"5cb1f66c2c363734fdd594cdb2eacf935402b0869d282db861010a7cfb0ce420"},"balance":"0"}]}]},"version":1,"chainId":"testnet-001","txHash":"3329CF8881290C66F6EAF46FC47ED2C398FB1868FE56C3AC7BAF2C2A32A6AFD4"}
```

**Comment by EVODelavega** [comment]
Running this test, we can in fact see the LP fees being transferred to the LP, but the transfer is buried under a mass of `MARGIN_LOW` and `BOND_PENALTY` transfers. The party simply doesn't have sufficient balance in this case. Here's some of the relevant transfers:

```
{
  "id": "215-5",
  "block": "442DD294DF657BC9F6F4CA11643C46290BEF8547CCAD571C0B34DB0C811C46A3",
  "type": "BUS_EVENT_TYPE_LEDGER_MOVEMENTS",
  "ledgerMovements": {
    "ledgerMovements": [
      {
        "entries": [
          {
            "fromAccount": {
              "assetId": "82b0674c519e97bd80bdf3d942e71f3004bdecc5a3362821aac40f32ce4f7423",
              "type": "ACCOUNT_TYPE_FEES_LIQUIDITY",
              "marketId": "be4a84643aa31c941302dec6450318e6fe7711114e0c28274292e28f3d86ef39"
            },
            "toAccount": {
              "assetId": "82b0674c519e97bd80bdf3d942e71f3004bdecc5a3362821aac40f32ce4f7423",
              "type": "ACCOUNT_TYPE_LP_LIQUIDITY_FEES",
              "owner": "e53a4a130467e9b55ac7cca9ecab4c6c7465657e7e50d1b1f754bff43f80a9c8",
              "marketId": "be4a84643aa31c941302dec6450318e6fe7711114e0c28274292e28f3d86ef39"
            },
            "amount": "359700000000000000000",
            "type": "TRANSFER_TYPE_LIQUIDITY_FEE_ALLOCATE",
            "timestamp": "1695141219115009332",
            "fromAccountBalance": "0",
            "toAccountBalance": "359700000000000000000"
          }
        ],
        "balances": [
          {
            "account": {
              "assetId": "82b0674c519e97bd80bdf3d942e71f3004bdecc5a3362821aac40f32ce4f7423",
              "type": "ACCOUNT_TYPE_LP_LIQUIDITY_FEES",
              "owner": "e53a4a130467e9b55ac7cca9ecab4c6c7465657e7e50d1b1f754bff43f80a9c8",
              "marketId": "be4a84643aa31c941302dec6450318e6fe7711114e0c28274292e28f3d86ef39"
            },
            "balance": "359700000000000000000"
          }
        ]
      }
    ]
  },
  "version": 1,
  "chainId": "testnet-001",
  "txHash": "442DD294DF657BC9F6F4CA11643C46290BEF8547CCAD571C0B34DB0C811C46A3"
}
{
  "id": "229-6",
  "block": "DDBAFA8053AAA7226F4C43D281DDCA1FEABB427CDA159BD02BE238DFD9A9F3A1",
  "type": "BUS_EVENT_TYPE_LEDGER_MOVEMENTS",
  "ledgerMovements": {
    "ledgerMovements": [
      {
        "entries": [
          {
            "fromAccount": {
              "assetId": "82b0674c519e97bd80bdf3d942e71f3004bdecc5a3362821aac40f32ce4f7423",
              "type": "ACCOUNT_TYPE_BOND",
              "owner": "e53a4a130467e9b55ac7cca9ecab4c6c7465657e7e50d1b1f754bff43f80a9c8",
              "marketId": "be4a84643aa31c941302dec6450318e6fe7711114e0c28274292e28f3d86ef39"
            },
            "toAccount": {
              "assetId": "82b0674c519e97bd80bdf3d942e71f3004bdecc5a3362821aac40f32ce4f7423",
              "type": "ACCOUNT_TYPE_INSURANCE",
              "marketId": "be4a84643aa31c941302dec6450318e6fe7711114e0c28274292e28f3d86ef39"
            },
            "amount": "48812500000000000000000",
            "type": "TRANSFER_TYPE_SLA_PENALTY_BOND_APPLY",
            "timestamp": "1695141228749872899",
            "fromAccountBalance": "48812500000000000000000",
            "toAccountBalance": "341687500000000000000000"
          }
        ],
        "balances": [
          {
            "account": {
              "assetId": "82b0674c519e97bd80bdf3d942e71f3004bdecc5a3362821aac40f32ce4f7423",
              "type": "ACCOUNT_TYPE_INSURANCE",
              "marketId": "be4a84643aa31c941302dec6450318e6fe7711114e0c28274292e28f3d86ef39"
            },
            "balance": "48812500000000000000000"
          }
        ]
      }
    ]
  },
  "version": 1,
  "chainId": "testnet-001",
  "txHash": "DDBAFA8053AAA7226F4C43D281DDCA1FEABB427CDA159BD02BE238DFD9A9F3A1"
}
{
  "id": "229-9",
  "block": "DDBAFA8053AAA7226F4C43D281DDCA1FEABB427CDA159BD02BE238DFD9A9F3A1",
  "type": "BUS_EVENT_TYPE_LEDGER_MOVEMENTS",
  "ledgerMovements": {
    "ledgerMovements": [
      {
        "entries": [
          {
            "fromAccount": {
              "assetId": "82b0674c519e97bd80bdf3d942e71f3004bdecc5a3362821aac40f32ce4f7423",
              "type": "ACCOUNT_TYPE_LP_LIQUIDITY_FEES",
              "owner": "e53a4a130467e9b55ac7cca9ecab4c6c7465657e7e50d1b1f754bff43f80a9c8",
              "marketId": "be4a84643aa31c941302dec6450318e6fe7711114e0c28274292e28f3d86ef39"
            },
            "toAccount": {
              "assetId": "82b0674c519e97bd80bdf3d942e71f3004bdecc5a3362821aac40f32ce4f7423",
              "type": "ACCOUNT_TYPE_INSURANCE",
              "marketId": "be4a84643aa31c941302dec6450318e6fe7711114e0c28274292e28f3d86ef39"
            },
            "amount": "359700000000000000000",
            "type": "TRANSFER_TYPE_SLA_PENALTY_LP_FEE_APPLY",
            "timestamp": "1695141228749872899",
            "fromAccountBalance": "0",
            "toAccountBalance": "342047200000000000000000"
          }
        ],
        "balances": [
          {
            "account": {
              "assetId": "82b0674c519e97bd80bdf3d942e71f3004bdecc5a3362821aac40f32ce4f7423",
              "type": "ACCOUNT_TYPE_INSURANCE",
              "marketId": "be4a84643aa31c941302dec6450318e6fe7711114e0c28274292e28f3d86ef39"
            },
            "balance": "359700000000000000000"
          }
        ]
      }
    ]
  },
  "version": 1,
  "chainId": "testnet-001",
  "txHash": "DDBAFA8053AAA7226F4C43D281DDCA1FEABB427CDA159BD02BE238DFD9A9F3A1"
}
{
  "id": "260-6",
  "block": "E2116B55DBE225F03036B28DD975C336822403339FB7B659A81A1CAB950E7C54",
  "type": "BUS_EVENT_TYPE_LEDGER_MOVEMENTS",
  "ledgerMovements": {
    "ledgerMovements": [
      {
        "entries": [
          {
            "fromAccount": {
              "assetId": "82b0674c519e97bd80bdf3d942e71f3004bdecc5a3362821aac40f32ce4f7423",
              "type": "ACCOUNT_TYPE_BOND",
              "owner": "e53a4a130467e9b55ac7cca9ecab4c6c7465657e7e50d1b1f754bff43f80a9c8",
              "marketId": "be4a84643aa31c941302dec6450318e6fe7711114e0c28274292e28f3d86ef39"
            },
            "toAccount": {
              "assetId": "82b0674c519e97bd80bdf3d942e71f3004bdecc5a3362821aac40f32ce4f7423",
              "type": "ACCOUNT_TYPE_INSURANCE",
              "marketId": "be4a84643aa31c941302dec6450318e6fe7711114e0c28274292e28f3d86ef39"
            },
            "amount": "24406250000000000000000",
            "type": "TRANSFER_TYPE_SLA_PENALTY_BOND_APPLY",
            "timestamp": "1695141249992982514",
            "fromAccountBalance": "24406250000000000000000",
            "toAccountBalance": "366453450000000000000000"
          }
        ],
        "balances": [
          {
            "account": {
              "assetId": "82b0674c519e97bd80bdf3d942e71f3004bdecc5a3362821aac40f32ce4f7423",
              "type": "ACCOUNT_TYPE_INSURANCE",
              "marketId": "be4a84643aa31c941302dec6450318e6fe7711114e0c28274292e28f3d86ef39"
            },
            "balance": "24406250000000000000000"
          }
        ]
      }
    ]
  },
  "version": 1,
  "chainId": "testnet-001",
  "txHash": "E2116B55DBE225F03036B28DD975C336822403339FB7B659A81A1CAB950E7C54"
}
{
  "id": "271-6",
  "block": "3833AEE8C3494987460CD0FAAEDBDC8C8C8A8EB15FABFC56FA49E2C6E0A474BC",
  "type": "BUS_EVENT_TYPE_LEDGER_MOVEMENTS",
  "ledgerMovements": {
    "ledgerMovements": [
      {
        "entries": [
          {
            "fromAccount": {
              "assetId": "82b0674c519e97bd80bdf3d942e71f3004bdecc5a3362821aac40f32ce4f7423",
              "type": "ACCOUNT_TYPE_BOND",
              "owner": "e53a4a130467e9b55ac7cca9ecab4c6c7465657e7e50d1b1f754bff43f80a9c8",
              "marketId": "be4a84643aa31c941302dec6450318e6fe7711114e0c28274292e28f3d86ef39"
            },
            "toAccount": {
              "assetId": "82b0674c519e97bd80bdf3d942e71f3004bdecc5a3362821aac40f32ce4f7423",
              "type": "ACCOUNT_TYPE_INSURANCE",
              "marketId": "be4a84643aa31c941302dec6450318e6fe7711114e0c28274292e28f3d86ef39"
            },
            "amount": "12203125000000000000000",
            "type": "TRANSFER_TYPE_SLA_PENALTY_BOND_APPLY",
            "timestamp": "1695141256453242732",
            "fromAccountBalance": "12203125000000000000000",
            "toAccountBalance": "378656575000000000000000"
          }
        ],
        "balances": [
          {
            "account": {
              "assetId": "82b0674c519e97bd80bdf3d942e71f3004bdecc5a3362821aac40f32ce4f7423",
              "type": "ACCOUNT_TYPE_INSURANCE",
              "marketId": "be4a84643aa31c941302dec6450318e6fe7711114e0c28274292e28f3d86ef39"
            },
            "balance": "12203125000000000000000"
          }
        ]
      }
    ]
  },
  "version": 1,
  "chainId": "testnet-001",
  "txHash": "3833AEE8C3494987460CD0FAAEDBDC8C8C8A8EB15FABFC56FA49E2C6E0A474BC"
}
{
  "id": "282-6",
  "block": "654F99EE6C7713ACC84DCBD35FF1353391D43D6B32E6F6FAA2EE9068B663015C",
  "type": "BUS_EVENT_TYPE_LEDGER_MOVEMENTS",
  "ledgerMovements": {
    "ledgerMovements": [
      {
        "entries": [
          {
            "fromAccount": {
              "assetId": "82b0674c519e97bd80bdf3d942e71f3004bdecc5a3362821aac40f32ce4f7423",
              "type": "ACCOUNT_TYPE_BOND",
              "owner": "e53a4a130467e9b55ac7cca9ecab4c6c7465657e7e50d1b1f754bff43f80a9c8",
              "marketId": "be4a84643aa31c941302dec6450318e6fe7711114e0c28274292e28f3d86ef39"
            },
            "toAccount": {
              "assetId": "82b0674c519e97bd80bdf3d942e71f3004bdecc5a3362821aac40f32ce4f7423",
              "type": "ACCOUNT_TYPE_INSURANCE",
              "marketId": "be4a84643aa31c941302dec6450318e6fe7711114e0c28274292e28f3d86ef39"
            },
            "amount": "6101562500000000000000",
            "type": "TRANSFER_TYPE_SLA_PENALTY_BOND_APPLY",
            "timestamp": "1695141263082942055",
            "fromAccountBalance": "6101562500000000000000",
            "toAccountBalance": "384758137500000000000000"
          }
        ],
        "balances": [
          {
            "account": {
              "assetId": "82b0674c519e97bd80bdf3d942e71f3004bdecc5a3362821aac40f32ce4f7423",
              "type": "ACCOUNT_TYPE_INSURANCE",
              "marketId": "be4a84643aa31c941302dec6450318e6fe7711114e0c28274292e28f3d86ef39"
            },
            "balance": "6101562500000000000000"
          }
        ]
      }
    ]
  },
  "version": 1,
  "chainId": "testnet-001",
  "txHash": "654F99EE6C7713ACC84DCBD35FF1353391D43D6B32E6F6FAA2EE9068B663015C"
}
```

TL;DR the party is bleeding balance

**Comment by gordsport** [comment]
@vega-paul - please can you re-test based on the above information and close if needed
